### PR TITLE
Validators + Encryption

### PR DESCRIPTION
**Pull request by KirbyNguyen**
Hey! I hope you don't mind me choosing this for a school project!
I have added an email validator and a strong password validator for when the user signs up. However, these are just functions. I haven't implemented them in the form yet. 
Additionally, I added encryption to the app. Before adding to the database, the app should encrypt the password. Additionally, when it retrieves the information from the database, it should also decrypt it.

### CODE REVIEW

File: ios/Podfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  File.foreach(generated_xcode_build_settings_path) do |line|
+    matches = line.match(/FLUTTER_ROOT\=(.*)/)
+    return matches[1].strip if matches
+  end
+  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
+end
+
+require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)
```
Comment (singhsaubhik) [inline ios/Podfile:26]: I am not sure, if these `ios/*` files are auto generated or has been added you. If added by you could you please explain a little bit about it, I am not a `IOS` developer.

File: lib/dao/dao.dart [hunk 2]
```diff
@@ -18,6 +19,14 @@ class DBHelper {
...
   static const _updatedAt = "updatedAt";
 
+  // Create a key for encryption
+  static final key = Key.fromUtf8('my 32 length key................');
+  // IV
+  static final iv = IV.fromLength(16);
+  // Create encrypter
+  static final encrypter =
+      Encrypter(AES(Key.fromUtf8('my 32 length key................')));
+
```
Comment (singhsaubhik) [inline lib/dao/dao.dart]: Can we try to make a `singleton` class named like `AppEncrypter` or something and use that here. So basic idea is that everything related `encryption` should be at one place

File: lib/dao/dao.dart [hunk 3]
```diff
@@ -83,11 +93,25 @@ class DBHelper {
...
+    row["password"] = encrypter.encrypt(row["password"], iv: iv).base64;
+    print("Encrypted password: " + row["password"]);
+    return row;
+  }
+
+  // Decrypt password in the row
+  // static Map<String, dynamic> decryptRow(Map<String, dynamic> row) {
+  //   row["password"] = encrypter.decrypt(row["password"], iv: iv);
+  //   return row;
+  // }
```
Comment (singhsaubhik) [inline lib/dao/dao.dart]: Remove this commented code

File: lib/provider/login_provider.dart [hunk 4]
```diff
@@ -28,7 +28,20 @@ class PasswordProvider extends ChangeNotifier {
...
+          DBHelper.encrypter.decrypt64(_list[index].password, iv: DBHelper.iv);
+    }
+
     _initialList = _list.toList();
+
+    // Decrypt the password in _initialList
+    // for (var index = 0; index < _initialList.length; index++) {
+    //   _initialList[index].password =
+    //       DBHelper.encrypter.decrypt64(_initialList[index].password, , iv: DBHelper.iv);
+    // }
```
Comment (singhsaubhik) [inline lib/provider/login_provider.dart]: Remove this commented code as well

**Review by singhsaubhik — CHANGES_REQUESTED** [review]
Hey, thank you so much for the contribution, I really appreciate. I have suggested some changes please try to address as many as possible.

Your validation codes looks really good. Thanks

File: ios/Podfile (hunk 1 above)
Comment (KirbyNguyen) [reply ios/Podfile:26]: Yes, this was autogenerated why I tried to run it on my iOS simulator

Comment (singhsaubhik) [reply ios/Podfile:26]: Ok, thanks for clarification

**Review by singhsaubhik — APPROVED** [review]

**Pull request merged**